utils/src/lib.rs

```diff
@@ -2,6 +2,8 @@
 
 use sha2::{Digest, Sha256};
 
+pub mod duration;
```

Comment:
Might want to do:
```
mod duration;
pub use duration::Extension;
```

Reply:
This actually would lose context that `Extension` is for duration, so instead you probably want to leave a `mod comment` at the top of `src/duration` (that is `!//`).